test/helpers/data_writer_helper.rb

```diff
@@ -0,0 +1,132 @@
+require "logger"
+require "thread"
+
+require "db_helper"
+require "ghostferry_helper"
+require "mysql2"
+
+module DataWriterHelper
+  def start_datawriter_with_ghostferry(dw, gf, &on_write)
+    gf.on_status(GhostferryHelper::Ghostferry::Status::READY) do
+      dw.start(&on_write)
+    end
+  end
+
+  def stop_datawriter_during_cutover(dw, gf)
+    gf.on_status(GhostferryHelper::Ghostferry::Status::ROW_COPY_COMPLETED) do
+      # At the start of the cutover phase, we have to set the database to
+      # read-only. This is done by stopping the datawriter.
+      dw.stop_and_join
+    end
+  end
+
+  class DataWriter
```

Comment:
Much of this is translated directly from data_writer.go. The goal is eventually we can use Ruby tests only.